src/ui/viewport/viewport.cpp

```diff
@@ -550,33 +548,83 @@ void Viewport::drawForeground(QPainter* painter, const QRectF& rect)
         painter->drawLine(o.toPointF(), p.first.toPointF());
     }
 
-    // Then add a text label in the lower-left corner
-    // giving mouse coordinates (if we're near an axis)
-    QPair<char, float> axis = getAxis();
-    QPointF mouse_pos = mapToScene(mapFromGlobal(QCursor::pos()));
-    if (!sceneRect().contains(mouse_pos))
-        axis.first = '\0';
+    // If we're on an axis (which means the viewport is showing mouse
+    // coordinates), force a redraw on mouse motion.
+    if (getAxis().first)
+    {
+        // Then add a text label in the lower-left corner
+        // giving mouse coordinates (if we're near an axis)
+        QPair<char, float> axis = getAxis();
+        QPointF mouse_pos = mapToScene(mapFromGlobal(QCursor::pos()));
+        if (!sceneRect().contains(mouse_pos))
+            axis.first = '\0';
+
+        auto p = sceneToWorld(mouse_pos);
+
+        QPointF a = sceneRect().bottomLeft() + QPointF(10, -25);
+        QPointF b = sceneRect().bottomLeft() + QPointF(10, -10);
+        painter->setPen(QColor(axis.second*200, axis.second*200, axis.second*200));
+        if (axis.first == 'z')
+        {
+            painter->drawText(a, QString("X: %1").arg(p.x()));
+            painter->drawText(b, QString("Y: %1").arg(p.y()));
+        }
+        else if (axis.first == 'y')
+        {
+            painter->drawText(a, QString("X: %1").arg(p.x()));
+            painter->drawText(b, QString("Z: %1").arg(p.z()));
+        }
+        else if (axis.first == 'x')
+        {
+            painter->drawText(a, QString("Y: %1").arg(p.y()));
+            painter->drawText(b, QString("Z: %1").arg(p.z()));
+        }
+
+    }
+
+    /* top left view info 'panel' */
+    painter->setPen(QColor(255, 255, 255));
+    QPointF top_left_info = sceneRect().topLeft();
 
-    auto p = sceneToWorld(mouse_pos);
+    /* display the currently hovered item or the number of overlaps */
+    int overlapping = 0;
+    QSet<Node*> used;
+    Node *active_node = NULL;
 
-    QPointF a = sceneRect().bottomLeft() + QPointF(10, -25);
-    QPointF b = sceneRect().bottomLeft() + QPointF(10, -10);
-    painter->setPen(QColor(axis.second*200, axis.second*200, axis.second*200));
-    if (axis.first == 'z')
+    for (auto i : items(_current_pos))
```

Comment:
I've added a `QList<ControlProxy*> Viewport::getProxiesAtPosition` and switched to using it in the mouse click event handler (924b747fdaa45f05fb51f99cf7a2c4b6d3a163d3) 

Want to switch to using it here?  You'll still have to call `getNode` to go from `ControlProxy` to `Node`, but it factors out the code that does the node deduplication.